Formula/ptpd.rb

```diff
@@ -0,0 +1,31 @@
+class Ptpd < Formula
+  desc "PTP daemon (PTPd) implementing the Precision Time Protocol (PTP) v2"
+  homepage "https://github.com/ptpd/ptpd"
+  url "https://github.com/ptpd/ptpd/archive/refs/tags/ptpd-2.3.1.tar.gz"
+  sha256 "267ad61d09d97069acec5d4878dceda20d0ddbebd27557d80230847848cee6c2"
+  license "BSD-2-Clause"
+  head "https://github.com/ptpd/ptpd.git", branch: "master"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
+  on_macos do
+    on_intel do
+      url "https://github.com/ptpd/ptpd.git", revision: "1ec9e650b03e6bd75dd3179fb5f09862ebdc54bf"
+    end
+    on_arm do
+      url "https://github.com/ptpd/ptpd.git", revision: "1ec9e650b03e6bd75dd3179fb5f09862ebdc54bf"
+    end
+  end
+
+  def install
+    system "autoreconf", "-vi"
+    system "./configure", *std_configure_args, "--disable-silent-rules"
+    system "make", "install"
+  end
+
+  test do
+    system "#{sbin}/ptpd2", "-v"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
The provided test in the project is to just run the daemon with an example conf file. However to do this we would need to know the ethernet interface name to update this conf file, and we would also need to send a SIGINT or something to the daemon to stop the test. Unfortunately this doesn't seem possible here, so I can't see any better options.

Reply:
How about checking if it fails as expected?

Reply:
Any command other than `-v` requires `ptpd` to be run as root, I have updated the test to check for this instead 👍